### PR TITLE
[SIG-3483] Keep marker cluster open when selecting containers

### PR DESCRIPTION
**Pull request by aditudorache**
This PR implements the missing functionality for keeping the cluster spyderfied when selecting containers. Also adds functionality to zoom to the bounds of the selected cluster marker when the map is zoomed out.

### CODE REVIEW

File: src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx [hunk 1]
```diff
@@ -98,16 +151,48 @@ export const ContainerLayer: FunctionComponent<DataLayerProps> = ({ featureTypes
...
+            const selectedLatLng = selectedCluster.current.getLatLng();
+            if (isEqual(latlng, selectedLatLng)) selectedCluster.current = undefined;
+            else selectedCluster.current = event.layer;
+          } else {
+            selectedCluster.current = event.layer;
+            selectedCluster.current.spiderfy();
+          }
+        } else {
+          // use this offset when zoomToBounds to keep the markers above the panel in mobile view
+          event.layer.zoomToBounds({ paddingTopLeft: [0, -300] });
```
Comment (MauritsR) [inline src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx:170]: The offset should only be applied on mobile (now it's always applied)

File: src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx [hunk 2]
```diff
@@ -98,16 +151,48 @@ export const ContainerLayer: FunctionComponent<DataLayerProps> = ({ featureTypes
...
+        }
+      });
+
+      /* istanbul ignore next */
+      if (selectedCluster.current) {
+        const selectedLatLng = selectedCluster.current.getLatLng();
+        const cluster = layerInstance.getLayers().find(layer => {
```
Comment (MauritsR) [inline src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx:177]: `(layerInstance.getLayers() as ClusterMarker[])` would remove the need for extra type casting later on

**Review by MauritsR — CHANGES_REQUESTED** [review]

File: src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx [hunk 3]
```diff
@@ -1,36 +1,87 @@
...
+  zoomToBounds: (options: any) => void;
+}
+
+export const getMarker = (parent: ClusterMarker | undefined, zoom: number): ClusterMarker | undefined => {
+  if (!parent) return undefined;
+  if (parent._zoom === zoom) return parent;
+  return getMarker(parent.__parent, zoom);
+};
+
+export const shouldSpiderfy = (layer: ClusterMarker, maxZoom?: number) => {
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx:35]: `spiderfy` is the term that leaflet uses. I'll keep this term for consistency.

File: src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx [hunk 4]
```diff
@@ -1,36 +1,87 @@
...
+  const cluster = layer;
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx:36]: Good point! Replaced.

File: src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx (hunk 1 above)
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx:170]: This is discussed and agreed with the UX designer

**Review by MauritsR — APPROVED** [review]

**Review by MauritsR — APPROVED** [review]

**Pull request merged**